src/bun.js/bindings/webcrypto/CryptoKeyOKPOpenSSL.cpp

```diff
@@ -294,26 +294,38 @@ ExceptionOr<Vector<uint8_t>> CryptoKeyOKP::exportPkcs8() const
 String CryptoKeyOKP::generateJwkD() const
 {
     ASSERT(type() == CryptoKeyType::Private);
+    if (namedCurve() == NamedCurve::Ed25519) {
+        ASSERT(m_exportKey);
+        return base64URLEncodeToString(*m_exportKey);
+    }
     return base64URLEncodeToString(m_data);
 }
 
-String CryptoKeyOKP::getEd25519PublicFromPrivate() const
+Span<const uint8_t> CryptoKeyOKP::ed25519PublicFromPrivate(KeyMaterial& privateKey)
 {
-    uint8_t publicKey[ED25519_PUBLIC_KEY_LEN];
-    uint8_t privateKey[ED25519_PRIVATE_KEY_LEN];
+    uint8_t public_key[ED25519_PUBLIC_KEY_LEN], private_key[ED25519_PRIVATE_KEY_LEN];
 
-    ED25519_keypair_from_seed(publicKey, privateKey, m_data.data());
+    ED25519_keypair_from_seed(public_key, private_key, privateKey.data());
 
-    return base64URLEncodeToString(Span<const uint8_t> { publicKey, sizeof(publicKey) });
+    return Span<const uint8_t> { public_key, sizeof(public_key) };
 }
 
-String CryptoKeyOKP::getX25519PublicFromPrivate() const
+Span<const uint8_t> CryptoKeyOKP::x25519PublicFromPrivate(KeyMaterial& privateKey)
 {
-    uint8_t publicKey[X25519_PUBLIC_VALUE_LEN];
+    uint8_t public_key[X25519_PUBLIC_VALUE_LEN];
```

Comment:
Isn't this a use-after-stack free? public_key and private key is statically allocated

Reply:
Instead, KeyMaterial& would need to be a reference holding the publicKey and privateKey